### PR TITLE
NMS-14423: Document allowed characters in service-names

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14423

### CODE REVIEW

**Comment by mfuhrmann** [comment]
Based on @deejgregor comment here https://issues.opennms.org/browse/NMS-1962?focusedCommentId=17135&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17135 I checked the type again. It is `varchar` now which allows 255 chars.

I wasn't able to find more information on allowed characters. I know I had issues in the past with poller names that broke the web UI in a node view or the availability views (category.xml, views.xml).
That's why I've listed only the ones that work for sure.

Not sure what else we could write here. Maybe a sentence like "If you use any other chars you might break some features that query or filter service names".

File: docs/modules/reference/pages/service-assurance/introduction.adoc [hunk 1]
```diff
@@ -6,6 +6,16 @@ You can configure them to allow complex monitoring to meet the needs of your env
 
 For information on how to extend the functionality of these monitors, see the xref:development:development.adoc#development[Development section] to contribute to the project.
 
+[[ref-service-assurance-monitors-name-definition]]
+== Name definitions
+
+Service names are subject to certain definitions:
+
+* Maximal length of 256 characters
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/introduction.adoc:14]:
```suggestion
* Maximum length of 256 characters
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Minor change

**Comment by Bonrob2** [comment]
Thanks for doing this, @mfuhrmann !

> Not sure what else we could write here. Maybe a sentence like "If you use any other chars you might break some features that query or filter service names".

I think adding a sentence like you suggest is a good idea.

File: docs/modules/reference/pages/service-assurance/introduction.adoc [hunk 2]
```diff
@@ -6,6 +6,16 @@ You can configure them to allow complex monitoring to meet the needs of your env
...
+* Alphapbetic characters +[Aa-Zz]+
+* Numerals +[0-9]+
```
Comment (emarsh1995) [inline docs/modules/reference/pages/service-assurance/introduction.adoc:16]:
```suggestion
* Alphanumeric characters +[Aa-Zz, 0-9]+
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
One minor change. Also, should this PR be targeting `foundation-2022` instead of `develop`?

**Comment by Bonrob2** [comment]
@mmahacek , Marcel is away, so could you also review so we can merge it for this week's release?

**Review by mmahacek — APPROVED** [review]
The only other comment I have would be to mention that spaces are not allowed. But I'm okay with this version.

**Comment by Bonrob2** [comment]
> The only other comment I have would be to mention that spaces are not allowed. But I'm okay with this version.

I'm going to need to rebase this branch anyway, so I'll add the comment about spaces.

**Comment by Bonrob2** [comment]
Closing this PR and opening a new one in order to rebase off foundation 2022 branch.

**Comment by Bonrob2** [comment]
Closing this PR and opening a new one in order to rebase off foundation-2022 branch: https://github.com/OpenNMS/opennms/pull/5160

**Comment by mfuhrmann** [comment]
Thanks for taking care!

@mmahacek 

> The only other comment I have would be to mention that spaces are not allowed. But I'm okay with this version.

Do we know what could happen in this case? Just asking because I have service names with spaces and :crossed_fingers: until now I never had problems.

**Comment by mmahacek** [comment]
You can't use `catinc` filters when there's spaces in names. I know there's other caveats but can't remember off hand.

**Pull request closed without merging**